domain/token/token_retriever.go

```diff
@@ -0,0 +1,57 @@
+package token
```

Comment:
I think we should add tests for this new package and file. This page helped me a lot: http://onsi.github.io/ginkgo/

If we could add a file header comment as well, that would be awesome

Reply:
@corinnekrych You missed this one too